katsdpbfingest/receiver.cpp

```diff
@@ -22,70 +26,25 @@
 // having this file depend on pybind11.
 namespace py = pybind11;
 
-bf_stream::~bf_stream()
-{
-    stop();
-}
-
-bf_stream::bf_stream(receiver &recv, const spead2::recv::stream_config &stream_config)
-    : spead2::recv::stream(recv.worker, stream_config),
-    recv(recv)
-{
-}
-
-void bf_stream::heap_ready(spead2::recv::live_heap &&heap)
-{
-    if (!heap.is_contiguous())
-        return;
-    recv.heap_ready(spead2::recv::heap(std::move(heap)));
-}
-
-void bf_stream::stop_received()
-{
-    spead2::recv::stream::stop_received();
-    recv.stop_received();
-}
-
-
-bf_raw_allocator::bf_raw_allocator(receiver &recv) : recv(recv)
-{
-}
-
-bf_raw_allocator::pointer bf_raw_allocator::allocate(std::size_t size, void *hint)
-{
-    std::uint8_t *ptr = nullptr;
-    if (hint)
-        ptr = recv.allocate(size, *reinterpret_cast<const spead2::recv::packet_header *>(hint));
-    if (ptr)
-        return pointer(ptr, deleter(shared_from_this(), (void *) std::uintptr_t(true)));
-    else
-        return spead2::memory_allocator::allocate(size, hint);
-}
-
-void bf_raw_allocator::free(std::uint8_t *ptr, void *user)
-{
-    if (!user)
-        delete[] ptr;
-}
-
 constexpr std::size_t receiver::window_size;
 constexpr int receiver::bf_raw_id;
 constexpr int receiver::timestamp_id;
 constexpr int receiver::frequency_id;
 
-slice receiver::make_slice()
+std::unique_ptr<slice> receiver::make_slice()
 {
-    slice s;
+    std::unique_ptr<slice> s{new slice};
```

Comment:
I can't recall seeing curlies like this before... Please remind me :-)

Reply:
I see slices are now accessed via pointers. Is that a spead2 change?

Reply:
For the curlies: https://www.geeksforgeeks.org/uniform-initialization-in-c/

Yes. spead2 uses `std::unique_ptr<spead2::recv::chunk>` in its ringbuffer - using a pointer allows for polymorphism (`slice` is a subclass of `spead2::recv::chunk`).